src/unittest/test_serialization.cpp

```diff
@@ -635,8 +635,39 @@ void TestSerialization::testBufReader()
 
 void TestSerialization::testFloatFormat()
 {
+	FloatType type = getFloatSerializationType();
 	u32 i;
 	f32 fs, fm;
+
+	// Check precision of float calculations on this platform
+	const std::unordered_map<f32, u32> float_results = {
+		{  0.0f, 0x00000000UL},
+		{  1.0f, 0x3F800000UL},
+		{ -1.0f, 0xBF800000UL},
+		{  0.1f, 0x3DCCCCCDUL},
+		{ -0.1f, 0xBDCCCCCDUL},
+		{ 1945329.25f, 0x49ED778AUL},
+		{ -23298764.f, 0xCBB1C166UL},
+		{  0.5f, 0x3F000000UL},
+		{ -0.5f, 0xBF000000UL}
+	};
+	for (const auto &v : float_results) {
+		i = f32Tou32Slow(v.first);
+		if (std::abs((s64)v.second - i) > 32) {
+			printf("Inaccurate float values on %.9g, expected 0x%X, actual 0x%X\n",
+				v.first, v.second, i);
+			UASSERT(false);
+		}
+		u32 i2 = f32Tou32Slow(u32Tof32Slow(i));
+		UASSERT(i == i2); // Lossy conversion
```

Comment:
This does not necessarily hold, especially if the platform has less precision in the mantissa. The number may not round-trip.

A suggested alternative to test the opposite direction is to convert `v.second` to float and check if the result is between `v.first * 0.999995f` and `v.first * 1.000005f` (but watch out for sign wrt comparison direction).